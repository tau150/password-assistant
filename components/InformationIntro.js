import * as React from 'react'
import Image from "next/image"
import { FormattedMessage } from 'react-intl'
import { Box, Heading, Flex, Text, Checkbox } from '@chakra-ui/react'

import { useStepsWizard } from '../contexts/stepWizardContext'
import headImage from '../public/images/head.png'
import boxImage from '../public/images/box.png'
import logo from '../public/images/key_openbank.png'

function InformationIntro(){
  const [checkedAge, setCheckedAge] = React.useState(false)

  const { setIsStepDisabled } = useStepsWizard()

  React.useEffect( () => {
    setIsStepDisabled(!checkedAge)
  }, [checkedAge])

  return (
    <Box as='section'>
        <Heading textAlign='center' as='h1' fontSize='3xl' color='brand.primary' mb={4}>
          <FormattedMessage id='welcomeTitle' defaultMessage='Welcome to your OpenClose Account'/>
        </Heading>
        <Box textAlign='center'>
          <Image width='70px' height='50px' src={logo} alt='openbank logo'/>
        </Box>
      <Box as='article' py={8} px={20}>
        <Heading as='h2' fontSize='3xl'>
          <FormattedMessage id='step1Title' defaultMessage='Create your Password Manager'/>
        </Heading>
        <Flex mt={8} w='100%'>
           <Flex w='50%' flexDir='column' alignItems='center'>
            <Image src={headImage} alt='head'/>
            <Text w='50%' textAlign='center'>
              <FormattedMessage id='descSave' defaultMessage='Save here all your passwords, data, or any information, forget paper notes and unprotected applications'/>
            </Text>
           </Flex>
           <Flex w='50%' flexDir='column' alignItems='center'>
            <Image src={boxImage} alt='vault'/>
            <Text w='50%' textAlign='center'>
              <FormattedMessage id='descCreate' defaultMessage='Create your master password: only you can access your secrets with it'/>
            </Text>
           </Flex>
        </Flex>
        <Heading mt={20} as='h3' fontSize='2xl'>
          <FormattedMessage id='howWorksTitle' defaultMessage='How it works'/>
        </Heading>
        <Text mt={6} maxW='60%'>
          <FormattedMessage
            id='howWorksDesc'
            defaultMessage='First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.'/>
        </Text>
        <Heading mt={20} as='h3' fontSize='2xl'>
          <FormattedMessage id='dataToSaveTitle' defaultMessage='What data can you save'/>
        </Heading>
        <Text mt={6} maxW='60%'>
          <FormattedMessage
            id='dataToSaveDesc'
            defaultMessage='For example, the number of your card, the PIN and PUK of your mobile phone, the serial number of one of your devices or any information that you need to have in a safe place.'/>
        </Text>
        <Heading mt={20} as='h3' fontSize='2xl'>
          <FormattedMessage id='howToContinue' defaultMessage='How to continue'/>
        </Heading>
        <Text mt={6} maxW='60%'>
          <FormattedMessage
            id='howToContinueDesc'
            defaultMessage='In order to continue you must be over 18 years old, so you will have to confirm it with the check below. Then you must choose your master password that meets the minimum security requirements, confirm it and, if you wish, indicate a recovery track.'/>
        </Text>
        <Checkbox mt={12} onChange={(e) => setCheckedAge(e.target.checked)}>
          <FormattedMessage id='ageConfirmation' defaultMessage='I confirm that I am at least 18 years old'/>
        </Checkbox>
      </Box>
    </Box>
  )
}

export default InformationIntro