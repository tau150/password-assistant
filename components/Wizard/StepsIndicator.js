import * as React from 'react'
import PropTypes from 'prop-types'
import { Badge, Box, Flex } from '@chakra-ui/react'

function StepsIndicator({stepsNumber, activeStep}){
  const arrayOfSteps = Array.from(Array(stepsNumber), (_, idx) => idx + 1)

  return (
    <Flex alignItems='center' position='relative'>
      {arrayOfSteps.map(number => (
        <Box key={number} position='relative'>
          {number !== stepsNumber && <Box left='20px' top='calc(50% - 2.5px)' zIndex='-1' position='absolute' as='span' w='100px' h='5px' bgColor='brand.accent'/>}
          <Badge
            fontSize='18px'
            w={activeStep === number ? "40px" : '30px'}
            h={activeStep === number ? "40px" : '30px'}
            borderRadius='50%'
            d='flex'
            justifyContent='center'
            alignItems='center'
            color='white'
            bgColor={activeStep === number ? "brand.secondary" : 'brand.accent'}
            boxShadow={activeStep === number ? "lg" : 'initial'}
            role="status"
            aria-label={`${number} ${activeStep === number ? 'is active' : ''}`}
            m='20px'
            position='relative'
            zIndex={3}
            key={number}>
              {number}
          </Badge>
        </Box>))}

    </Flex>
  )
}

StepsIndicator.propTypes = {
  stepsNumber: PropTypes.number,
  activeStep: PropTypes.number
}

export default StepsIndicator