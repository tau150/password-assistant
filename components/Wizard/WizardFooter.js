import * as React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { Flex } from '@chakra-ui/react'

import Button from '../ui/Button'

function WizardFooter({ handlePrevStep, handleNextStep, isStepDisabled, isPrevDisabled}){
  return (
    <Flex as='footer' borderTop='1px' py={12} px={20} borderColor='gray.200' alignItems='center' justifyContent='space-between'>
      <Button aria-label='prev-button' variant="secondary" disabled={isPrevDisabled} onClick={handlePrevStep}>
        <FormattedMessage  id='cancel' defaultMessage='Cancel'/>
      </Button>
      <Button aria-label='next-button' variant="primary" disabled={isStepDisabled} onClick={handleNextStep}>
        <FormattedMessage id='next' defaultMessage='Next'/>
      </Button>
    </Flex>
  )
}

WizardFooter.propTypes = {
  handlePrevStep: PropTypes.func.isRequired,
  handleNextStep: PropTypes.func.isRequired,
  isStepDisabled: PropTypes.bool,
  isPrevDisabled: PropTypes.bool
}

export default WizardFooter