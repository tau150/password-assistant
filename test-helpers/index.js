
import * as React from 'react'
import {   render as rtlRender } from '@testing-library/react'
import { IntlProvider } from 'react-intl';
import { ChakraProvider } from "@chakra-ui/react"

import theme from '../styles/theme';
import intlENmessages from '../content/locale/en.json'

function AppProviders({children}){
  return (
    <IntlProvider
      locale='en'
      messages={intlENmessages}
      onError={() => null}>
      <ChakraProvider theme={theme}>
        {children}
      </ChakraProvider>
    </IntlProvider>
  )
}


export function render(ui, ...renderOptions){
  const returnValue = {
    ...rtlRender(ui, {wrapper: AppProviders, ...renderOptions}),
  }
  return returnValue
}
