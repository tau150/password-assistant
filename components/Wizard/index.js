import * as React from 'react'
import PropTypes from 'prop-types'
import { Box, Flex } from '@chakra-ui/react'

import { StepWizardContext } from '../../contexts/stepWizardContext'
import StepsIndicator from './StepsIndicator'
function Wizard({children}){
  const [currentStep, setCurrentStep] = React.useState(0)
  const [isStepDisabled, setIsStepDisabled] = React.useState(false)

  const stepsNumberBasedOnChildren = React.Children.count(children);

  const handleNextStep = async () => {
    setCurrentStep( prev => prev += 1)
  }

  const handlePrevStep = async () => {
    setCurrentStep( prev => prev -= 1)
  }

  const arrayOfChildren = React.Children.toArray(children);
  const currentContent = arrayOfChildren[currentStep];
  const isLastStep = currentStep + 1 === stepsNumberBasedOnChildren

  const value = {
    currentStep,
    numberOfSteps: stepsNumberBasedOnChildren,
    setCurrentStep,
    setIsStepDisabled,
    isStepDisabled,
    handleNextStep,
    handlePrevStep,
    isLastStep
  }

  return (
    <StepWizardContext.Provider value={value}>
      <Box border='1px solid brand.accent' boxShadow='lg'>
        <Flex as='header' position='relative' zIndex='2' h='80px' bgColor='gray.200' alignItems='center' justifyContent='center' boxShadow='md'>
          <StepsIndicator stepsNumber={stepsNumberBasedOnChildren} activeStep={currentStep + 1}/>
        </Flex>
        <Box p='4'>
          {currentContent}
        </Box>
      </Box>
    </StepWizardContext.Provider>
  )
}


Wizard.propTypes = {
  children: PropTypes.node,
}

export default Wizard