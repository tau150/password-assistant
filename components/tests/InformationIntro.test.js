import * as React from 'react'
import { screen } from '@testing-library/dom'
import { render } from '../../test-helpers'
import userEvent from '@testing-library/user-event';

import InformationIntro from '../InformationIntro'
import Wizard from '../Wizard'


function IntroWrapper() {
  return (
    <Wizard>
      <InformationIntro />
    </Wizard>
  )
}

describe('Information Intro test', () => {
  it('renders Information Intro', () => {
    expect(render(<IntroWrapper />)).toMatchSnapshot()
  })

  it('toggle checkbox', () => {
    render(<IntroWrapper />)

    let checkbox = screen.getByRole('checkbox', {name: 'age'})
    expect(checkbox.checked).toEqual(false)

    userEvent.click(checkbox)
    checkbox = screen.getByRole('checkbox', {name: 'age'})
    expect(checkbox.checked).toEqual(true)
  })
})

