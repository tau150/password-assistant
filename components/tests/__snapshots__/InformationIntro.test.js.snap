// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Intro test renders Information Intro 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-l50gwe"
      >
        <header
          class="css-62ow20"
        >
          <div
            class="css-14s4iyf"
          >
            <div
              class="css-79elbk"
            >
              <span
                aria-label="1 is active"
                class="chakra-badge css-1aiadu4"
                role="status"
              >
                1
              </span>
            </div>
          </div>
        </header>
        <div
          class="css-h94677"
        >
          <section
            class="css-0"
          >
            <h1
              class="chakra-heading css-xjra6q"
            >
              Welcome to your OpenClose Account
            </h1>
            <article
              class="css-ejhs3n"
            >
              <h2
                class="chakra-heading css-1n7pa12"
              >
                Create your Password Manager
              </h2>
              <div
                class="css-j02tnc"
              >
                <div
                  class="css-13i0eth"
                >
                  <img
                    alt="test"
                    src=""
                  />
                  <p
                    class="chakra-text css-1hrwr3f"
                  >
                    Save here all your passwords, data, or any information, forget paper notes and unprotected applications
                  </p>
                </div>
                <div
                  class="css-13i0eth"
                >
                  <img
                    alt="test"
                    src=""
                  />
                  <p
                    class="chakra-text css-1hrwr3f"
                  >
                    reate your master password: only you can access your secrets with it
                  </p>
                </div>
              </div>
              <h3
                class="chakra-heading css-14dj72c"
              >
                How it works
              </h3>
              <p
                class="chakra-text css-1pnzxoz"
              >
                First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.
              </p>
              <h3
                class="chakra-heading css-14dj72c"
              >
                What data can you save
              </h3>
              <p
                class="chakra-text css-1pnzxoz"
              >
                For example, the number of your card, the PIN and PUK of your mobile phone, the serial number of one of your devices or any information that you need to have in a safe place.
              </p>
              <h3
                class="chakra-heading css-14dj72c"
              >
                How to continue
              </h3>
              <p
                class="chakra-text css-1pnzxoz"
              >
                In order to continue you must be over 18 years old, so you will have to confirm it with the check below. Then you must choose your master password that meets the minimum security requirements, confirm it and, if you wish, indicate a recovery track.
              </p>
              <label
                class="chakra-checkbox css-h4lg0k"
              >
                <input
                  aria-label="age"
                  class="chakra-checkbox__input"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="checkbox"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="chakra-checkbox__control css-xxkadm"
                />
                <span
                  class="chakra-checkbox__label css-1sgc0qu"
                >
                  I confirm that I am at least 18 years old
                </span>
              </label>
            </article>
            <footer
              class="css-1ry2npb"
            >
              <button
                aria-label="prev-button"
                class="chakra-button css-1t0rcyz"
                disabled=""
                type="button"
              >
                Cancel
              </button>
              <button
                aria-label="next-button"
                class="chakra-button css-1t0rcyz"
                disabled=""
                type="button"
              >
                Next
              </button>
            </footer>
          </section>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-l50gwe"
    >
      <header
        class="css-62ow20"
      >
        <div
          class="css-14s4iyf"
        >
          <div
            class="css-79elbk"
          >
            <span
              aria-label="1 is active"
              class="chakra-badge css-1aiadu4"
              role="status"
            >
              1
            </span>
          </div>
        </div>
      </header>
      <div
        class="css-h94677"
      >
        <section
          class="css-0"
        >
          <h1
            class="chakra-heading css-xjra6q"
          >
            Welcome to your OpenClose Account
          </h1>
          <article
            class="css-ejhs3n"
          >
            <h2
              class="chakra-heading css-1n7pa12"
            >
              Create your Password Manager
            </h2>
            <div
              class="css-j02tnc"
            >
              <div
                class="css-13i0eth"
              >
                <img
                  alt="test"
                  src=""
                />
                <p
                  class="chakra-text css-1hrwr3f"
                >
                  Save here all your passwords, data, or any information, forget paper notes and unprotected applications
                </p>
              </div>
              <div
                class="css-13i0eth"
              >
                <img
                  alt="test"
                  src=""
                />
                <p
                  class="chakra-text css-1hrwr3f"
                >
                  reate your master password: only you can access your secrets with it
                </p>
              </div>
            </div>
            <h3
              class="chakra-heading css-14dj72c"
            >
              How it works
            </h3>
            <p
              class="chakra-text css-1pnzxoz"
            >
              First, you must create a different password for your electronic belongings. You will not be able to recover your password, so remember it well.
            </p>
            <h3
              class="chakra-heading css-14dj72c"
            >
              What data can you save
            </h3>
            <p
              class="chakra-text css-1pnzxoz"
            >
              For example, the number of your card, the PIN and PUK of your mobile phone, the serial number of one of your devices or any information that you need to have in a safe place.
            </p>
            <h3
              class="chakra-heading css-14dj72c"
            >
              How to continue
            </h3>
            <p
              class="chakra-text css-1pnzxoz"
            >
              In order to continue you must be over 18 years old, so you will have to confirm it with the check below. Then you must choose your master password that meets the minimum security requirements, confirm it and, if you wish, indicate a recovery track.
            </p>
            <label
              class="chakra-checkbox css-h4lg0k"
            >
              <input
                aria-label="age"
                class="chakra-checkbox__input"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value=""
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-xxkadm"
              />
              <span
                class="chakra-checkbox__label css-1sgc0qu"
              >
                I confirm that I am at least 18 years old
              </span>
            </label>
          </article>
          <footer
            class="css-1ry2npb"
          >
            <button
              aria-label="prev-button"
              class="chakra-button css-1t0rcyz"
              disabled=""
              type="button"
            >
              Cancel
            </button>
            <button
              aria-label="next-button"
              class="chakra-button css-1t0rcyz"
              disabled=""
              type="button"
            >
              Next
            </button>
          </footer>
        </section>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
